source/game.roomhandler.adagency.bmx

```diff
@@ -786,6 +790,10 @@ Type RoomHandler_AdAgency Extends TRoomHandler
 
 	Method RefillBlocksMode1()
 		TLogger.Log("AdAgency.RefillBlocks", "RefillBlocksMode1.", LOG_DEBUG)
+		If Not refillData
+			refillData = new SAdContractRefillData
```

Comment:
Structs existieren IMMER ... es gibt also kein "If Not refillData" und auch kein "Field refillData:SAdContractRefillData = Null".

Behelf waere entweder ein "Wrapper" (`Type TAdContractRefillData; Field s:SadContractRefillData; End Type`) oder eine Hilfsvariable: `Field refillDataIsSet:int = False` welche dann entsprechend gesetzt und geprueft werden kann (der Integer erspart uns das "Objekt" ...)

Reply:
OK. Danke für den Hinweis. In dem Fall ist ja nur wichtig, dass bei jedem Spielstart (Laden oder neues Spiel) einmalig das update aufgerufen wird. Ein separater Wrapper oder eine update-Variable sollten nicht nötig sein.

Reply:
Unnötige Null-Behandlung ist behoben.

Reply:
Unnötige Null-Behandlung ist behoben.